bin/get-amy.py

```diff
@@ -38,15 +35,25 @@ def main():
     }
 
     # Save.
-    with open(output_file, 'w') as writer:
-        yaml.dump(config, writer, encoding='utf-8', allow_unicode=True)
+    try:
+        with open(output_file, 'w') as writer:
+            yaml.dump(config, writer, encoding='utf-8', allow_unicode=True)
+    except TypeError:
+        # Output_file isn't a string, so it's a file.
+        yaml.dump(config, output_file, encoding='utf-8', allow_unicode=True)
 
 
-def fetch_info(base_url, url):
+def fetch_info(base_url, url, tags=None):
     '''Download and save data.'''
 
     address = base_url + url
 
+    # Consider filtering by tags.
+    if tags:
+        query_params = [urllib.parse.urlencode({'tag': tag}) for tag in tags]
+        query = '&'.join(query_params)
```

Comment:
On Fri, Jan 01, 2016 at 02:20:12PM -0800, Piotr Banaszkiewicz wrote:
> 2. multiple items of the same name are Django's way of handling
> forms (see for example AMY's code for bulk upload) and GET query
> params: instead of `dict`, Django creates a `QueryDict`, which
> provides `.getlist()` method

You don't have to use .getlist, you can use something like:

  tags = request.GET.get('tag', '').split(',')

But without an AMY change like that, I don't see a way around the ‘&’
joining you're using here.  In the end, it doesn't really matter which
way you go.